sdk/core/src/migration/credentials/index.ts

```diff
@@ -0,0 +1,163 @@
+    this.keysService = keysService
+    this.platformCryptographyTools = platformCryptographyTools
+    this.apiKey = apiKey
+    this.api = new ApiService(this.baseUrl, {
+      'Api-Key': apiKey,
+      'X-SDK-Version': version,
+    })
+    this.bloomDid = bloomDid
+  }
+
+  /**
+   * Fetches authorization token to `vault-migration-service`
+   */
+  private async getAuth(): Promise<string> {
+    if (this.auth && !this.didAuthService.isTokenExpired(this.auth, Date.now())) {
+      return this.auth
+    }
+
+    const requestToken = await this.didAuthService.pullDidAuthRequestToken()
+    this.auth = await this.didAuthService.createDidAuthResponseToken(requestToken)
+
+    return this.auth
+  }
+
+  /**
+   * Copy of the private method of `AffinidiVaultStorageService`
+   * @param credentials
+   * @private
+   */
+  private async encryptCredentials(credentials: any[]): Promise<vcMigrationList[]> {
+    const publicKeyBuffer = this.keysService.getOwnPublicKey()
+    const privateKeyBuffer = this.keysService.getOwnPrivateKey()
+    const encryptedCredentials: vcMigrationList[] = []
+
+    for (const credential of credentials) {
+      let credentialId = credential?.id
+      if (!isW3cCredential(credential)) {
+        credentialId = credential?.data?.id
+      }
+
+      const credentialIdHash = await this.platformCryptographyTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+      const typeHashes = []
+      if (isW3cCredential(credential)) {
+        for (const credentialType of credential.type) {
+          const typeHash = await this.platformCryptographyTools.computePersonalHash(privateKeyBuffer, credentialType)
+          typeHashes.push(typeHash)
+        }
+      }
+
+      const cyphertext = await this.platformCryptographyTools.encryptByPublicKey(publicKeyBuffer, credential)
+
+      encryptedCredentials.push({
+        id: credentialIdHash,
+        types: typeHashes,
+        payload: cyphertext,
+        bloomVaultIndex: credential.bloomId,
+      })
+    }
+
+    return encryptedCredentials
+  }
+
+  /**
+   * Initiate VCs migration on bloom-vault VC READ event.
+   * @param credentials
+   */
+  async runMigration(credentials: any[]): Promise<void> {
+    const encryptedVCs = await this.encryptCredentials(credentials)
+    await this.runMigrationByChunk(encryptedVCs, 100)
+  }
+
+  /**
+   * Divides given encrypted VCs list into chunks and run migration
+   * @param encryptedVCs
+   * @param chunk
+   */
+  async runMigrationByChunk(encryptedVCs: any[], chunk: number): Promise<void> {
+    for (let i = 0, L = encryptedVCs.length; i < L; i += chunk) {
+      const chunkedVCList = encryptedVCs.slice(i, i + chunk)
+      await this.migrateCredentials(chunkedVCList)
+    }
+  }
+
+  /**
+   * Gets migration status for user with given token
+   */
+  async getMigrationStatus(): Promise<{ status: string }> {
+    const token = await this.getAuth()
+    const url = 'api/v1/migrationStatus'
+    return this.api.execute(
```

Comment:
Please just add a new service for that into `internal-api-clients` instead of reimplementing all the base functionality such as manual `fetch`ing and did-auth.

It looks like most of the code changes in this PR is just a reimplementation of generic functionality that is already implemented in `internal-api-clients`.